web/autograder_service.go

```diff
@@ -855,6 +855,22 @@ func (s *AutograderService) IsEmptyRepo(ctx context.Context, in *pb.RepositoryRe
 		}
 		return nil, status.Errorf(codes.FailedPrecondition, "group repository does not exist or not empty")
 	}
-
 	return &pb.Void{}, nil
 }
+
+// GetStudentForDiscord fetches student name and ID by the course name and student's GitHub login
+// Access policy: Course teachers
+func (s *AutograderService) GetStudentForDiscord(ctx context.Context, in *pb.DiscordRequest) (*pb.DiscordResponse, error) {
+	usr, err := s.getCurrentUser(ctx)
```

Comment:
Will the Discord bot need to authenticate with GitHub to use the method??

Reply:
Well, is it not reasonable to expect authentication with a user who is also a course teacher? Because otherwise we would just expose a method returning a personal information on any user that could be called by literally anyone. 
And GitHub authentication is the only option we support for now.

Reply:
Yes, we should definitely have authentication; we just need to provide an example, e.g. in the form of a test case or something, so that it becomes easy to integrate into the discord bot.